### PR TITLE
Fixes Null Reference + Qdel Issue in Runechat

### PR DESCRIPTION
**Pull request by bobbah**
## About The Pull Request
Fixed a runtime that could occur in runechat if the spans list / extra classes list was null.
Fixed an issue with signal registration that would cause a runtime when trying to qdel self after parent is qdeling.

## Why It's Good For The Game
no runtimes

## Changelog
:cl: bobbahbrown
fix: Fixed unstyled messages potentially causing null reference exceptions in runechat.
fix: Fixed a client's qdeletion causing runtimes on chat messages they see.
/:cl:

### CODE REVIEW

File: code/datums/chatmessage.dm [hunk 1]
```diff
@@ -93,15 +93,17 @@
...
 
 	// Append radio icon if from a virtual speaker
-	if (extra_classes.Find("virtual-speaker"))
+	if (extra_classes?.Find("virtual-speaker"))
 		var/image/r_icon = image('icons/UI_Icons/chat/chat_icons.dmi', icon_state = "radio")
 		text =  "\icon[r_icon]&nbsp;" + text
 
 	// We dim italicized text to make it more distinguishable from regular text
-	var/tgt_color = extra_classes.Find("italics") ? target.chat_color_darkened : target.chat_color
+	var/tgt_color = extra_classes?.Find("italics") ? target.chat_color_darkened : target.chat_color
```
Comment (Rohesie) [inline code/datums/chatmessage.dm:106]: Do `/datum/chatmessage/proc/generate_image(text, atom/target, mob/owner, list/extra_classes = list(), lifespan)` instead

File: code/datums/chatmessage.dm [hunk 2]
```diff
@@ -171,6 +173,8 @@
 	// Check for virtual speakers (aka hearing a message through a radio)
 	var/atom/movable/originalSpeaker = speaker
 	if (istype(speaker, /atom/movable/virtualspeaker))
+		if (!spans)
+			spans = list()
```
Comment (Rohesie) [inline code/datums/chatmessage.dm:177]: Do `spans = spans ? spans.Copy() : list()` instead. Handle it early.

File: code/datums/chatmessage.dm [hunk 3]
```diff
@@ -96,12 +102,12 @@
 		extra_classes |= "small"
 
 	// Append radio icon if from a virtual speaker
-	if (extra_classes.Find("virtual-speaker"))
+	if (extra_classes?.Find("virtual-speaker"))
```
Comment (Rohesie) [inline code/datums/chatmessage.dm:105]:
```suggestion
	if (extra_classes.Find("virtual-speaker"))
```
No need for this anymore.

File: code/datums/chatmessage.dm [hunk 4]
```diff
@@ -96,12 +102,12 @@
...
 		var/image/r_icon = image('icons/UI_Icons/chat/chat_icons.dmi', icon_state = "radio")
 		text =  "\icon[r_icon]&nbsp;" + text
 
 	// We dim italicized text to make it more distinguishable from regular text
-	var/tgt_color = extra_classes.Find("italics") ? target.chat_color_darkened : target.chat_color
+	var/tgt_color = extra_classes?.Find("italics") ? target.chat_color_darkened : target.chat_color
```
Comment (Rohesie) [inline code/datums/chatmessage.dm:110]:
```suggestion
	var/tgt_color = extra_classes.Find("italics") ? target.chat_color_darkened : target.chat_color
```
Nor this.

File: code/datums/chatmessage.dm [hunk 5]
```diff
@@ -68,7 +74,7 @@
 /datum/chatmessage/proc/generate_image(text, atom/target, mob/owner, list/extra_classes, lifespan)
```
Comment (Rohesie) [inline code/datums/chatmessage.dm:74]:
```suggestion
/datum/chatmessage/proc/generate_image(text, atom/target, mob/owner, list/extra_classes = list(), lifespan)
```

File: code/datums/chatmessage.dm [hunk 6]
```diff
@@ -68,7 +74,7 @@
...
 	// Register client who owns this message
 	owned_by = owner.client
-	RegisterSignal(owned_by, COMSIG_PARENT_QDELETING, .proc/qdel, src)
+	RegisterSignal(owned_by, COMSIG_PARENT_QDELETING, .proc/parent_qdel)
```
Comment (Rohesie) [inline code/datums/chatmessage.dm:77]:
```suggestion
	RegisterSignal(owned_by, COMSIG_PARENT_QDELETING, .proc/pn_parent_qdel)
```
Last suggestion before merge. Make this clear, else it implies the proc qdels the parent.

File: code/datums/chatmessage.dm [hunk 7]
```diff
@@ -55,6 +55,12 @@
 	message = null
 	return ..()
 
+/**
+  * Calls qdel on the chatmessage when its parent is deleted, used to register qdel signal
+  */
+/datum/chatmessage/proc/parent_qdel()
```
Comment (Rohesie) [inline code/datums/chatmessage.dm:61]:
```suggestion
/datum/chatmessage/proc/pn_parent_qdel()
```

**Comment by Archanial** [comment]
If timer subsystem failed because of that, this pr should close #50829 #50828

**Comment by bobbahbrown** [comment]
@Archanial 
> If timer subsystem failed because of that, this pr should close #50829 #50828 

it failed because of something with pipenet updates apparently, unrelated to this afaik

**Pull request merged**